test-avocado/cockpit.py

```diff
@@ -0,0 +1,179 @@
+
+        # SELinux messages to ignore
+        "(audit: )?type=1403 audit.*",
+        "(audit: )?type=1404 audit.*",
+
+        # Hmm
+        "request timed out, closing",
+        "(audit: )?type=1400 .* name=\"machine-info\".*",
+        "pam_lastlog\\(cockpit:session\\): unable to open /var/log/lastlog: No such file or directory"
+    ]
+
+    def allow_journal_messages(self, *patterns):
+        """Don't fail if the journal containes a entry matching the given regexp"""
+        for p in patterns:
+            self.allowed_messages.append(p)
+
+    def allow_restart_journal_messages(self):
+        self.allow_journal_messages("Error receiving data: Connection reset by peer",
+                                    "g_dbus_connection_real_closed: Remote peer vanished with error: Underlying GIOStream returned 0 bytes on an async read \\(g-io-error-quark, 0\\). Exiting.",
+                                    "g_dbus_connection_real_closed: Remote peer vanished with error: Error sending message: Broken pipe \\(g-io-error-quark, 44\\). Exiting.",
+                                    # HACK: https://bugzilla.redhat.com/show_bug.cgi?id=1141137
+                                    "localhost: bridge program failed: Child process killed by signal 9")
+
+    def journal_messages(self, syslog_ids, log_level):
+        """Return interesting journal messages"""
+
+        # Journald does not always set trusted fields like
+        # _SYSTEMD_UNIT or _EXE correctly for the last few messages of
+        # a dying process, so we filter by the untrusted but reliable
+        # SYSLOG_IDENTIFIER instead
+
+        matches = " ".join(map(lambda id: "SYSLOG_IDENTIFIER=" + id, syslog_ids))
+
+        # Some versions of journalctl terminate unsuccessfully when
+        # the output is empty.  We work around this by ignoring the
+        # exit status and including error messages from journalctl
+        # itself in the returned messages.
+
+        cmd = "journalctl 2>&1 -c'%s' -o cat -p %d %s" % (self.journal_start, log_level, matches)
+        out = process.run(cmd, shell=True, ignore_status=True)
+        messages = out.stdout.splitlines()
+        if len(messages) == 1 and "Cannot assign requested address" in messages[0]:
+            # No messages
+            return [ ]
+        else:
+            return messages
+
+    def audit_messages(self, type_pref):
+        cmd = "journalctl -c'%s' -o cat SYSLOG_IDENTIFIER=kernel 2>&1 | grep 'type=%s.*audit' || true" % (self.journal_start, type_pref)
+        out = process.run(cmd, shell=True)
+        messages = out.stdout.splitlines()
+        if len(messages) == 1 and "Cannot assign requested address" in messages[0]:
+            messages = [ ]
+        return messages
+
+    def check_journal_messages(self):
+        """Check for unexpected journal entries."""
+        syslog_ids = [ "cockpitd", "cockpit-ws", "cockpit-session" ]
+        messages = self.journal_messages(syslog_ids, 5)
```

Comment:
"Because that's how we have always done it." :-)  It's a valid question, but not related to the move to Avocado, I'd say.